c2rust-ast-builder/src/builder.rs

```diff
@@ -161,6 +161,23 @@ fn punct_box<T, P: Default>(x: Vec<Box<T>>) -> Punctuated<T, P> {
     Punctuated::from_iter(x.into_iter().map(|x| *x))
 }
 
+fn comma_separated<T, F>(items: Vec<T>, f: F) -> TokenStream
+where
+    F: Fn(&mut TokenStream, T),
+{
+    let mut tokens = TokenStream::new();
+    let mut first = true;
+    for item in items {
+        if !first {
+            TokenTree::Punct(Punct::new(',', Spacing::Alone)).to_tokens(&mut tokens);
+        } else {
+            first = false;
+        }
+        f(&mut tokens, item);
+    }
+    tokens
+}
```

Comment:
```suggestion
fn comma_separated<I, T>(items: I) -> TokenStream
where
    I: Iterator<Item = T>,
    T: ToTokens + Clone,
{
    let items = items.map(|items| items.to_token_stream());
    let comma = TokenTree::Punct(Punct::new(',', Spacing::Alone)).into_token_stream();
    intersperse(items, comma).collect()
}
```
If you add `itertools` to `c2rust-ast-builder` (which we already use for `c2rust-transpile`), this could be a lot simpler.

Reply:
if it is ok to add new dependencies crate, i will add it tomorrow.   cannot wait for https://github.com/rust-lang/rust/issues/79524 to become stable😃

Reply:
```
$ cargo build
  Downloaded pathdiff v0.2.3
  Downloaded cfg-if v1.0.3
  ....
  Downloaded linux-raw-sys v0.9.4
  Downloaded 114 crates (12.7 MB) in 31.19s (largest was `linux-raw-sys` at 2.3 MB)
error: package `clap v4.5.47` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.65.0-nightly
```

just simply adding itertools in c2rust-ast-builder, and running usual `cargo update` + `cargo build` fails when somehow clap suddenly upgraded to `4.5` 🤣. maybe I will just wait until https://github.com/immunant/c2rust/pull/1227 landed